src/Utils.cpp

```diff
@@ -468,11 +469,81 @@ utils::escapeBlacklistedHtml(const QString &rawStr)
 }
 
 QString
-utils::markdownToHtml(const QString &text)
-{
-        const auto str      = text.toUtf8();
-        const char *tmp_buf = cmark_markdown_to_html(str.constData(), str.size(), CMARK_OPT_UNSAFE);
+utils::markdownToHtml(const QString &text, bool rainbowify)
+{
+        const auto str = text.toUtf8();
+        cmark_node *const node =
+          cmark_parse_document(str.constData(), str.size(), CMARK_OPT_UNSAFE);
+
+        if (rainbowify) {
+                // create iterator over node
+                cmark_iter *iter = cmark_iter_new(node);
+
+                cmark_event_type ev_type;
+
+                // First loop to get total text length
+                int textLen = 0;
+                while ((ev_type = cmark_iter_next(iter)) != CMARK_EVENT_DONE) {
+                        cmark_node *cur = cmark_iter_get_node(iter);
+                        // only text nodes (no code or semilar)
+                        if (cmark_node_get_type(cur) != CMARK_NODE_TEXT)
+                                continue;
+                        // count up by length of current node's text
+                        textLen += strlen(cmark_node_get_literal(cur));
+                }
+
+                // create new iter to start over
+                cmark_iter_free(iter);
+                iter = cmark_iter_new(node);
+
+                // Second loop to rainbowify
+                int charIdx = 0;
+                while ((ev_type = cmark_iter_next(iter)) != CMARK_EVENT_DONE) {
+                        cmark_node *cur = cmark_iter_get_node(iter);
+                        // only text nodes (no code or semilar)
+                        if (cmark_node_get_type(cur) != CMARK_NODE_TEXT)
+                                continue;
+
+                        // get text in current node
+                        const char *tmp_buf = cmark_node_get_literal(cur);
+                        std::string nodeText(tmp_buf);
+                        // create buffer to append rainbow text to
+                        std::string buf;
+                        for (int i = 0; i < nodeText.length(); i++) {
```

Comment:
nodeText.length() returns a size_t, which isn't directly compatible with an int.  You can change i to a std::size_t to avoid this issue in CI.  This also works better with 'at', as 'at' assumed a size_t instead of an int (so you're actually casting the data type implicitly there).